src/main/java/com/expediagroup/hiveberg/IcebergSerDe.java

```diff
@@ -51,6 +64,40 @@ public void initialize(@Nullable Configuration configuration, Properties propert
     }
   }
 
+  private Table findTable(Configuration conf, Properties properties) throws IOException {
```

Comment:
This is the same code as in the InputFormat isn't it? Can you move it out into a Util class that both call? I think the idea is for this to be shared across both InputFormats based on the discussion in the other PR and will follow a similar approach.

Reply:
yeah, i'll refactor this out

Reply:
Changes made based on the comments :)